### PR TITLE
Feature/save all

### PR DESCRIPTION
**Pull request by LasyIsLazy**
Then:
readlater
Now:
readlater2
readlater3

Update:

1. Fix ellipse: long titles have difference length.
2. Support save all tabs: Save all tabs in current tab.
3. Move url to <a title="x">: When hover on title, there will be a title and url below title.

### CODE REVIEW

**Comment by LasyIsLazy** [comment]
I hope these updates will help to this extension.

File: popup.html [hunk 1]
```diff
@@ -10,8 +10,9 @@
 <body>
     <div class="spinner"></div>
     <div class="linkList">
-        <div>
+        <div class="buttonContainer">
             <button class="saveButton" type="button">Save Tab</button>
+            <button class="saveButton saveAllButton" type="button">Save All Tabs</button>
```
Comment (timleland) [inline popup.html]: Change this to have an id="saveAllButton" so that you do not get click event for saveButton

File: popup.html [hunk 2]
```diff
@@ -10,8 +10,9 @@
 <body>
     <div class="spinner"></div>
     <div class="linkList">
-        <div>
+        <div class="buttonContainer">
             <button class="saveButton" type="button">Save Tab</button>
```
Comment (timleland) [inline popup.html]: Change this to have an id="saveButton" and update the javascript to target the id not the class

**Review by timleland — CHANGES_REQUESTED** [review]
Thanks for submitting this improvement!

**Comment by LasyIsLazy** [comment]
Any plan for publishing a new version?

File: popup.html (hunk 2 above)
Comment (timleland) [inline popup.html]: Add and id for "saveButton" and switch **. saveButton** to use **#saveButton**

Right now the click event will be bound to both buttons

**Review by timleland — CHANGES_REQUESTED** [review]

**Review by timleland — APPROVED** [review]
Looks good thanks!

**Pull request merged**